tests/org.eclipse.ui.tests/Eclipse UI Tests/org/eclipse/ui/tests/dialogs/SearchPatternAuto.java

```diff
@@ -111,66 +91,42 @@ public void testPatternMatch2() {
 	}
 
 	/**
-	 * Tests camelCase match functionality.
-	 * Every string starting with an upperCase character should be recognized as camelCase pattern match rule.
-	 * Result for "CD" SearchPattern should be similar to regexp pattern "C[^A-Z]*D.*".
-	 * If pattern contains only upperCase characters, result also contains all prefix match elements.
+	 * Tests camelCase match functionality. Every string starting with an upperCase
+	 * character should be recognized as camelCase pattern match rule. If there is
+	 * no camelCase match, a simple substring match is tried as a fallback.
 	 */
 	@Test
 	public void testCamelCaseMatch() {
-		Pattern pattern = Pattern.compile("C[^A-Z]*D.*");
-		Pattern pattern2 = Pattern.compile("CD.*", Pattern.CASE_INSENSITIVE);
-		assertMatches("CD", SearchPattern.RULE_CAMELCASE_MATCH, pattern, pattern2);
+		Pattern pattern = Pattern.compile("B[^A-Z]*C.*");
+		Pattern fallbackPattern = Pattern.compile("BC.*", Pattern.CASE_INSENSITIVE);
+		assertMatches("BC", SearchPattern.RULE_CAMELCASE_MATCH, pattern, fallbackPattern);
 	}
 
 	/**
 	 * Tests camelCase match functionality.
-	 * Every string starting with an upperCase character should be recognized as camelCase pattern match rule.
-	 * Result for "AbCd " SearchPattern should be similar to regexp pattern "Ab[^A-Z]*Cd[^A-Z]*".
 	 */
 	@Test
-	public void testCamelCaseForcedEndMatch() {
-		Pattern pattern = Pattern.compile("Ab[^A-Z]*Cd[^A-Z]*");
-		assertMatches("AbCd ", SearchPattern.RULE_CAMELCASE_MATCH, pattern);
-		// alternative ending character:
-		assertMatches("AbCd<", SearchPattern.RULE_CAMELCASE_MATCH, pattern);
+	public void testCamelCaseChangingCaseMatch() {
```

Comment:
Why are those tests modified?
It would be better to keep them, to emphasize that previous use-cases are not broken; and to add extra ones.

Reply:
@mickaelistria, I see your point. The reason behind the changes is that I found it more "friendly" to have similar sets of tests for original vs. new way of search, so they are more easily comparable.

So I slightly amended the original tests, so that they cover the various scenarios in a better / broader way. For example this one - testing `"AbCd "` - missed the check that for example "`ZzAbCd`" won't be found by the original search (if I remember it correctly, all the generated `resources` basically start with `A`). Of course, I can revert some of the changes as you propose, but I'm not sure if it is worth it...

Reply:
@mickaelistria, so is it OK from you under the described circumstances?